internal/cypress/common.go

```diff
@@ -0,0 +1,57 @@
+package cypress
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/saucelabs/saucectl/internal/github"
+)
+
+// IsCypressVersionAvailable checks if the requested version is available on cloud or docker.
+func IsCypressVersionAvailable(version string) (DockerAvailability, CloudAvailability bool, err error) {
```

Comment:
@FriggaHel @alexplischke what are the case scenarios we check against GitHub API? is it every time we try to start a new test? what happens with the developer workflow if Github is down? is it blocking saucectl to run tests?

Reply:
> is it every time we try to start a new test?

Yup, every time a project is run!

> what happens with the developer workflow if Github is down? is it blocking saucectl to run tests?

Aye, currently this would block the execution.

❗ Now that I think about it, we don't really need to perform a docker availability check against github at all (not for the docker mode test case at least). That may only be necessary if you wanted to determine the "latest" version via github, but nobody actually asked for that feature (feature creep).
We can simply continue to get the version from docker hub, as we normally do and if that's not available based on what the user has specified, we fail. We don't need the overhead of a github check for that.

❗ ❗ As I dig down the rabbit hole further, there is technically no need to perform a github check at all. We could do this check inside test-composer as well. Of course we'd have to maintain a (simple) list of what versions are available (if we wanted to catch them inside test-composer before the request hits other services), but that's fairly simple to do maintenance wise, requires no additional calls to be made by saucectl and completely avoids a reliance on github.

Reply:
So to confirm the changes:
- Cloud: Add endpoint in test-composer checking the available versions in cloud.
- Docker: Rely on docker pull like before

Right @alexplischke @vrunoa  ?

Reply:
We don't need to add any new endpoint at all :)

So right now, if you were to provide a non-existant version, the request would fail when checked by platforms. That's totally fine. If we wanted to catch that earlier, like in test-composer for instance, we could perform the version check right there on a job start request (this doesn't require a new endpoint, we already have the job start endpoint). However, it only costs us one service jump where the check is currently performed, so having yet another pre-check in test-composer doesn't buy us much (we aren't wasting any resources).

As of this moment I'd say that there is actually nothing that needs to be done 👍